lib/xnvme_spec_pp.c

```diff
@@ -421,3 +421,63 @@ xnvme_spec_znd_type_str(enum xnvme_spec_znd_type eval)
 
 	return "ENOSYS";
 }
+
+int
+xnvme_spec_ctrlr_bar_fpr(FILE *stream, struct xnvme_spec_ctrlr_bar *bar, int opts)
+{
+	int wrtn = 0;
+
+	switch (opts) {
+	case XNVME_PR_TERSE:
+		wrtn += fprintf(stream, "# ENOSYS: opts(%x)", opts);
+		return wrtn;
+
+	case XNVME_PR_DEF:
+	case XNVME_PR_YAML:
+		break;
+	}
+
+	wrtn += fprintf(stream, "xnvme_spec_ctrlr_bar:");
+	if (!bar) {
+		wrtn += fprintf(stream, "~\n");
+		return wrtn;
+	}
+
+	wrtn += fprintf(stream, "\n");
+
+	wrtn += fprintf(stream, "  cap: 0x%016" PRIx64 "\n", bar->cap);
+	wrtn += fprintf(stream, "  vs: %" PRIu32 "\n", bar->vs);
+	wrtn += fprintf(stream, "  intms: %" PRIu32 "\n", bar->intms);
+	wrtn += fprintf(stream, "  intmc: %" PRIu32 "\n", bar->intmc);
+	wrtn += fprintf(stream, "  cc: %" PRIu32 "\n", bar->cc);
+	wrtn += fprintf(stream, "  rsvd24: %" PRIu32 "\n", bar->rsvd24);
```

Comment:
Does it make sense to print the reserved bits?

Reply:
hehe, no, not really, that trickled in removing it. Also, I am adding a note on the "feature-completeness" of this pretty-printer in the commit message.